src/ui/util/TableConfig.ts

```diff
@@ -0,0 +1,93 @@
+import { idb } from "../../worker/db";
```

Comment:
A file in the ui folder should not be importing anything from worker. Either make two separate versions of TableConfig, one for ui and one for worker, or make one base class in common and extend that for use in the worker.

Reply:
I'm not sure it makes sense to share the same class for ui and worker... seems every function is only called in one of those places.